source/extensions/filters/http/on_demand/BUILD

```diff
@@ -19,17 +19,22 @@ envoy_cc_library(
         "//envoy/event:dispatcher_interface",
         "//envoy/http:filter_interface",
         "//envoy/server:filter_config_interface",
+        "//envoy/upstream:cluster_manager_interface",
         "//source/common/common:assert_lib",
         "//source/common/common:enum_to_int",
+        "//source/common/config:xds_resource_lib",
         "//source/common/http:codes_lib",
         "//source/common/http:header_map_lib",
+        "//source/extensions/filters/http:well_known_names",
+        "@envoy_api//envoy/extensions/filters/http/on_demand/v3:pkg_cc_proto",
     ],
 )
 
 envoy_cc_extension(
     name = "config",
     srcs = ["config.cc"],
     hdrs = ["config.h"],
+    visibility = ["//visibility:public"],
```

Comment:
Is this needed only for tests?
If so, using something similar to this is better https://github.com/envoyproxy/envoy/blob/e8cd3164b77825d97c809694e88d050a20c3e52b/source/extensions/filters/http/ip_tagging/BUILD#L36-L39

Reply:
Yeah. Something must have changed since last time I tried to run the tests. Now, without those changes, the tests were failing because the on-demand extension was not registered.

Thanks for pointing me in the right direction.

Reply:
Changed to use `extra_visibility`. Let's see if it does the trick too.

Reply:
Moved the integration test to `test/extensions/filters/http/on_demand` so this visibility stuff is not needed any more. Dropped.